sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -184,9 +187,16 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
 
     val cls = DataSource.lookupDataSource(source)
     if (classOf[DataSourceV2].isAssignableFrom(cls)) {
-      val options = new DataSourceV2Options(extraOptions.asJava)
+      val dataSource = cls.newInstance()
+      val options = dataSource match {
+        case cs: SessionConfigSupport =>
+          val confs = withSessionConfig(cs.keyPrefix, sparkSession.sessionState.conf)
```

Comment:
just call `Utils. withSessionConfig` here. Don't import `Utils._` at the beginning